sphinx/ext/autosummary/__init__.py

```diff
@@ -329,7 +329,12 @@ def get_items(self, names):
 
             # -- Grab the signature
 
-            sig = documenter.format_signature()
+            try:
+                sig = documenter.format_signature(show_annotation=False)
```

Comment:
But this handles `TypeError` to support custom documenters. So I believe this does not break compatibility.